wpilibc/src/main/native/cpp/Commands/CommandGroup.cpp

```diff
@@ -165,32 +165,45 @@ void CommandGroup::_Execute() {
         break;
 
       case CommandGroupEntry::kSequence_BranchPeer:
+        // Start executing a parallel command and advance to next entry in group
         m_currentCommandIndex++;
-        entry.m_command->Start();
+        entry->m_command->Start();
         break;
 
       case CommandGroupEntry::kSequence_BranchChild:
         m_currentCommandIndex++;
-        CancelConflicts(entry.m_command);
-        entry.m_command->StartRunning();
+
+        /* Causes scheduler to skip children of current command which require
+         * the same subsystems as it
+         */
+        CancelConflicts(entry->m_command);
+        entry->m_command->StartRunning();
+
+        // Add current command entry to list of children of this group
         m_children.push_back(entry);
         break;
     }
   }
 
   // Run Children
-  for (auto iter = m_children.begin(); iter != m_children.end();) {
-    entry = *iter;
-    Command* child = entry.m_command;
-    if (entry.IsTimedOut()) child->_Cancel();
+  for (auto entry : m_children) {
```

Comment:
This needs to be `auto&` for the `entry = nullptr` to work.  Just plain `auto` makes a value copy.